clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerInterceptors.java

```diff
@@ -76,7 +78,8 @@ public ProducerInterceptors(List<ProducerInterceptor<K, V>> interceptors) {
      *
      * This method does not throw exceptions. Exceptions thrown by any of interceptor methods are caught and ignored.
      *
-     * @param metadata The metadata for the record that was sent (i.e. the partition and offset). Null if an error occurred.
+     * @param metadata The metadata for the record that was sent (i.e. the partition and offset).
+     *                 If an error occurred, metadata will only contain valid topic and maybe partition.
```

Comment:
We haven't thought about this option. Yes, this sounds like a bit cleaner implementation. @junrao what is your opinion? If we add RecordSendException class that will contain original exception, topic, and nullable partition. We could possibly use it later in Callback?